File: AIClient_JAVA/AIClient_JAVA/src/event/GameStartEvent.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** GameStartEvent.java
 ** Event that happen when a game start
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

import world.World;

public class GameStartEvent extends IngoingEvent {
	private int numberOfTeam, numberOfCharacter;
	private int[] teamIDs;

	@Override
	public void execute() {
		World.getInstance().startGame(numberOfTeam, numberOfCharacter, teamIDs);
	}

	@Override
	public boolean fillArguments(String message) {
		String[] headerParts = message.split(SEPARATOR, 3);

		numberOfTeam = convertCharToNumeral(headerParts[0]);
		numberOfCharacter = convertCharToNumeral(headerParts[1]);

		if (numberOfTeam < 1 || numberOfCharacter < 1) {
			return false;
		}

		//Plus one for the sender id that we don't use
		String[] ids = headerParts[2].split(SEPARATOR, numberOfTeam+1);
		for (int index = 0; index < ids.length-1; ++index) {
			teamIDs[index] = convertCharToNumeral(ids[index]);
			if (teamIDs[index] == 0) {
				return false;
			}
		}

		return false;
	}
}

File: AIClient_Python/src/event/GameStartEvent.py
'''
Created on Dec 20, 2014

@author: samuel
'''
from event.IngoingEvent import IngoingEvent
from aiclient.Singleton import Singleton
from world.World import World

class GameStartEvent(IngoingEvent):

    def __init__(self):
        self.numberOfTeam = 0
        self.numberOfCharacter = 0
        self.teamsIds = []

    def fillArguments(self, string):
        numberParts = string.split(":", 2)
        self.numberOfTeam = int(numberParts[0])
        self.numberOfCharacter = int(numberParts[1])
        
        # The last number is the game client id that we don't need
        teamIds = numberParts[2].split(":", self.numberOfTeam)
        for index in range(0, self.numberOfTeam):
            self.teamsIds.insert(index, int(teamIds[index]))
            
    def execute(self):
        world = Singleton(World)
        world.startGame(self.numberOfTeam, self.numberOfCharacter, self.teamsIds)